File: src/main/java/com/uap/control_tickets/services/interfaces/AdministrativoService.java
package com.uap.control_tickets.services.interfaces;

import com.uap.control_tickets.dto.administrativo.AdministrativoDetalleDto;
import com.uap.control_tickets.dto.administrativo.AdministrativoDto;
import com.uap.control_tickets.dto.administrativo.PrevisualizacionAdmCsvDto;
import com.uap.control_tickets.dto.estudiante.ImportacionResultadoDto;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/** Contrato del CRUD de Administrativo + carga masiva por CSV. */
public interface AdministrativoService {
    List<AdministrativoDetalleDto> listar();
    AdministrativoDetalleDto obtener(Long idAdministrativo);
    AdministrativoDetalleDto crear(AdministrativoDto dto);
    void eliminar(Long idAdministrativo);

    /** Carga masiva desde CSV. Columnas: codigo administrativo, nombre completo, ci. */
    ImportacionResultadoDto importarCsv(MultipartFile archivo);

    /** Lee el CSV y cuenta qué pasaría, SIN escribir nada (para la vista previa). */
    PrevisualizacionAdmCsvDto previsualizarCsv(MultipartFile archivo);
}

File: frontend/src/api/administrativo.service.ts
// Capa de API de Administrativos (CRUD + importación CSV).
import http from '@/api/http'
import type {
  AdministrativoDetalleDto,
  AdministrativoDto,
  PrevisualizacionAdmCsvDto,
} from '@/types/administrativo.type'
import type { ImportacionResultadoDto } from '@/types/estudiante.type'

export function listarAdministrativos() {
  return http.get<AdministrativoDetalleDto[]>('/administrativos/listar').then((r) => r.data)
}

export function crearAdministrativo(dto: AdministrativoDto) {
  return http.post<AdministrativoDetalleDto>('/administrativos/crear', dto).then((r) => r.data)
}

export function eliminarAdministrativo(idAdministrativo: number) {
  return http.delete('/administrativos/eliminar', { params: { idAdministrativo } })
}

export function importarAdministrativosCsv(archivo: File) {
  const fd = new FormData()
  fd.append('archivo', archivo)
  return http.post<ImportacionResultadoDto>('/administrativos/importar', fd).then((r) => r.data)
}

/** Corre el mismo parser que la importación real pero sin tocar la BD (vista previa). */
export function previsualizarAdministrativosCsv(archivo: File) {
  const fd = new FormData()
  fd.append('archivo', archivo)
  return http
    .post<PrevisualizacionAdmCsvDto>('/administrativos/previsualizar', fd)
    .then((r) => r.data)
}
